blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/JSONMetricsContainer.java

```diff
@@ -132,8 +137,22 @@ public void setTtlInSeconds(int ttlInSeconds) {
         }
 
         @JsonIgnore
-        public boolean isValid() {
-            return true;
+        public List<String> getValidationErrors() {
+
+            long current = System.currentTimeMillis();
+
+            List<String> errors = new ArrayList<String>();
+
+            // collectionTime is an optional parameter
+            if ( collectionTime != null
+                    && (collectionTime > current + futureDiff) )
+                errors.add( "'" + metricName + "': 'collectionTime' '" + collectionTime + "' is more than '" + futureDiff + "' milliseconds into the future." );
+
+            if ( collectionTime != null
+                    && (collectionTime < current - pastDiff) )
+                errors.add( "'" + metricName + "': 'collectionTime' '" + collectionTime + "' is more than '" + pastDiff + "' milliseconds into the past." );
+
+            return errors;
```

Comment:
Isnt there a standard way of handling json validation? Like something jackson provides? I didnt spend much time on it but there is http://hibernate.org/validator/documentation/getting-started/ which I believe we could use. What do you guys think?